deploy/chart/templates/daemonset.yaml

```diff
@@ -35,6 +35,15 @@ spec:
             {{- if .Values.linodegoDebug }}
             - --linodego-debug={{ .Values.linodegoDebug }}
             {{- end }}
+            {{- if .Values.routeController }}
+            - --enable-route-controller=true
+            - --vpc-name={{ .Values.routeController.vpcName }}
```

Comment:
is there a way for the ccm to figure this out automatically?

Reply:
Thats a good question. If we change the logic to always fetch instance ips and vpc ips, build a cache and check if instance is present in both, then we can say this particular instance is in VPC. However, when route_controller calls listRoutes, it will not know which instance is part of particular VPC and will list all routes configured on all instances in an account. This will result route_controller to send delete for routes of instances it doesn't expect in the cluster. It might be possible that we wait for createRoute call for an instance and then get its interfaces and that will tell us which vpc its part of, and then use the result going further, but I feel we might hit some edge cases with it. There are inbuilt flags in [cloud-provider](https://github.com/kubernetes/cloud-provider/blob/master/config/types.go#L71) but those haven't been useful, so I had to introduce this new one. Detecting it automatically would be so nice to have. Maybe something we can keep an eye on and if we find a way in future, switch to it.